### ISSUE TITLE
model.train(False) affects gradient tracking?

### ISSUE THREAD

**Issue by MaverickMeerkat** [pytorch/tutorials#2230]
In this tutorial here it says in the comment that "# We don't need gradients on to do reporting". From what I understand the train flag only affects layers such as dropout and batch-normalization. Does it also affect gradient calculations, or is this comment wrong?

https://github.com/pytorch/tutorials/blob/6bd30cf214bf541a1c5d35cc45d10a381f57af1b/beginner_source/introyt/trainingyt.py#L293

cc @suraj813

**Comment by chedatomasz** [comment]
This comment, and the accompanying code, is wrong. A more correct pattern is shown at https://pytorch.org/tutorials/beginner/basics/optimization_tutorial.html#full-implementation, although this one misses the train/eval switches and thus does not generalize to models with batchnorm and dropout.

**Comment by chedatomasz** [comment]
Related issues: #2083 #1756 #208
